src/Common/src/Interop/Windows/sspicli/SecuritySafeHandles.cs

```diff
@@ -464,141 +429,100 @@ internal abstract partial class SafeDeleteContext : SafeHandle
+                        fixed (char* namePtr = punyCode)
+                        {
+                            errorCode = MustRunInitializeSecurityContext(
+                                            ref inCredentials,
+                                            isContextAbsent,
+                                            (byte*)(((object)targetName == (object)dummyStr) ? null : namePtr),
+                                            inFlags,
+                                            endianness,
+                                            &inSecurityBufferDescriptor,
+                                            refContext,
+                                            ref outSecurityBufferDescriptor,
+                                            ref outFlags,
+                                            outFreeContextBuffer);
+                        }
 
-                    if (NetEventSource.IsEnabled) NetEventSource.Info(null, "Marshalling OUT buffer");
+                        if (NetEventSource.IsEnabled) NetEventSource.Info(null, "Marshalling OUT buffer");
 
-                    // Get unmanaged buffer with index 0 as the only one passed into PInvoke.
-                    outSecBuffer.size = outUnmanagedBuffer.cbBuffer;
-                    outSecBuffer.type = outUnmanagedBuffer.BufferType;
-                    if (outSecBuffer.size > 0)
-                    {
-                        outSecBuffer.token = new byte[outSecBuffer.size];
-                        Marshal.Copy(outUnmanagedBuffer.pvBuffer, outSecBuffer.token, 0, outSecBuffer.size);
-                    }
-                    else
-                    {
-                        outSecBuffer.token = null;
+                        // Get unmanaged buffer with index 0 as the only one passed into PInvoke.
+                        outSecBuffer.size = outUnmanagedBuffer.cbBuffer;
+                        outSecBuffer.type = outUnmanagedBuffer.BufferType;
+                        outSecBuffer.token = outSecBuffer.size > 0 ?
+                            new Span<byte>((byte*)outUnmanagedBuffer.pvBuffer, outUnmanagedBuffer.cbBuffer).ToArray() :
+                            null;
                     }
                 }
             }
             finally
             {
-                if (pinnedInBytes != null)
-                {
-                    for (int index = 0; index < pinnedInBytes.Length; index++)
-                    {
-                        if (pinnedInBytes[index].IsAllocated)
-                        {
-                            pinnedInBytes[index].Free();
-                        }
-                    }
-                }
-                if (pinnedOutBytes.IsAllocated)
-                {
-                    pinnedOutBytes.Free();
-                }
-
-                if (outFreeContextBuffer != null)
-                {
-                    outFreeContextBuffer.Dispose();
-                }
+                outFreeContextBuffer?.Dispose();
```

Comment:
`using`

Reply:
Can't do that in this case: it's being written to inside the try block in various places.